pkg/batcher/createfleet.go

```diff
@@ -19,24 +19,25 @@ import (
 	"fmt"
 	"time"
 
-	"github.com/aws/aws-sdk-go/aws"
-	"github.com/aws/aws-sdk-go/service/ec2"
-	"github.com/aws/aws-sdk-go/service/ec2/ec2iface"
+	"karpenter-provider-aws/pkg/aws"
+	"github.com/aws/aws-sdk-go-v2/aws"
+	"github.com/aws/aws-sdk-go-v2/service/ec2"
+	"github.com/aws/aws-sdk-go-v2/service/ec2/types"
 	"sigs.k8s.io/controller-runtime/pkg/log"
 )
 
 type CreateFleetBatcher struct {
 	batcher *Batcher[ec2.CreateFleetInput, ec2.CreateFleetOutput]
 }
 
-func NewCreateFleetBatcher(ctx context.Context, ec2api ec2iface.EC2API) *CreateFleetBatcher {
+func NewCreateFleetBatcher(ctx context.Context, awsClient awsapi.AWSAPI) *CreateFleetBatcher {
```

Comment:
I think this shouldn't change significantly with the interface changes that I'm recommending